File: java/src/main/java/io/fivexer/sdk/model/SupervisorEntry.java
package io.fivexer.sdk.model;

/**
 * Where a supervisor can sign in from. Unauthenticated, and deliberately takes no workspace id — echoing one back would turn it into an existence oracle.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class SupervisorEntry {
    private String consoleUrl;

    public String getConsoleUrl() { return consoleUrl; }
}

File: php/src/Model/SupervisorEntry.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Where a supervisor can sign in from. Unauthenticated, and deliberately takes no workspace id — echoing one back would turn it into an existence oracle.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class SupervisorEntry
{
    public function __construct(
        public readonly ?string $consoleUrl,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            consoleUrl: isset($data['consoleUrl']) ? (string) $data['consoleUrl'] : null,
        );
    }
}
